### ISSUE TITLE
Failed to apply optimize_transpose error when converting a two-layer keras model

### ISSUE THREAD

**Issue by mazeltovlee** [onnx/tensorflow-onnx#1903]
**Describe the bug**
Although I find many closed issues talking about the `Failed to apply optimize_*` problem, I think it may still be a unique issue that is unsolved since the root cause may be different. In short, I was building a model using keras and converting it to onnx using `tf2onnx`, during the job, I notice this error message even though the ONNX model is successfully generated. I simplify the model as follows:
```
import keras
# This is the buggy model built from scratch
layer_stack = [
  keras.layers.Conv2D(10, kernel_size=(3,3), strides=(2,2)),
  keras.layers.BatchNormalization(axis=-2),
]
keras_input = keras.layers.Input((6,6,512))
layer_input = keras_input
for layer in layer_stack:
  y = layer(layer_input)
  layer_input = y
model = keras.models.Model(keras_input, y)
model.summary()
model.save("temp.h5")
```

**To Reproduce**
You can directly reproduce the issue through this link:
https://colab.research.google.com/drive/1qhZREJ9c3wIfFTkO6aRKMHQXGUhwDA91?usp=sharing

**Screenshots**
<img width="1246" alt="image" src="https://user-images.githubusercontent.com/91937711/161237250-9d201446-63fc-456d-b7ca-0822d2d6f3f1.png">

Not sure if it can help, but I find out that, when I change the `axis` of BatchNormalization layer from `-2` to `-1`, no optimization error will be thrown.

**Comment by fatcat-z** [comment]
I tried your code with tensorflow==2.7.1 and keras==2.7.0, it works well whil we set axis=-2. Could you please take another try?

For tensorflow 2.8.0, it looks like the tf frozen graph is different than previous versions which caused the issue you met above.

**Comment by mazeltovlee** [comment]
Hi, I have tried on tensorflow==2.7.1 and keras==2.7.0. The optimization error still insists when I set axis=-2. The tf2onnx's version is 1.9.3.
Since I reproduce the bug in the colab, not sure if it is the colab's problem.
Please see this link for the code: 
https://colab.research.google.com/drive/1qhZREJ9c3wIfFTkO6aRKMHQXGUhwDA91?usp=sharing

**Comment by fatcat-z** [comment]
> Hi, I have tried on tensorflow==2.7.1 and keras==2.7.0. The optimization error still insists when I set axis=-2. The tf2onnx's version is 1.9.3. Since I reproduce the bug in the colab, not sure if it is the colab's problem. Please see this link for the code: https://colab.research.google.com/drive/1qhZREJ9c3wIfFTkO6aRKMHQXGUhwDA91?usp=sharing

Got it, I'm using the tf2onnx nightly build which can work with tf 2.7.1 and keras 2.7.0 well. Please try it or wait for the next release.

**Comment by mazeltovlee** [comment]
Thanks for your fix! 
May I know if you are going to fix this issue on the latest tf (2.8.0)?

**Comment by fatcat-z** [comment]
> Thanks for your fix! May I know if you are going to fix this issue on the latest tf (2.8.0)?

No plan for this yet.
